### PR TITLE
Add a test for side-effects

### PR DESCRIPTION
**Pull request by mrm007**
This allows as to see what side-effects we need to keep when we ship compiled Braid (v32).

### CODE REVIEW

File: site/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "react-use": "^17.4.0",
     "recoil": "^0.7.2",
     "sku": "11.7.0",
-    "webpack": "^5.0.0"
+    "webpack": "^5.72.1"
```
Comment (mrm007) [inline site/package.json:58]: This is the same version already in the lockfile.

File: packages/consumer/__snapshots__/build.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`build side-effects 1`] = `
```
Comment (mrm007) [inline packages/consumer/__snapshots__/build.test.ts.snap:3]: Having these as reference will be of great help in Crackle https://github.com/seek-oss/crackle/blob/6c0f1bc76167a3a8c418bffc30e82879395b4ce7/packages/core/src/package-utils/bundle.ts#L46-L59

File: packages/consumer/build.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import path from 'path';
+import glob from 'fast-glob';
+import webpack from 'webpack';
+import braidPkg from 'braid-design-system/package.json'; // eslint-disable-line no-restricted-imports
+
+const { SideEffectsFlagPlugin } = webpack.optimize;
+
+describe('build', () => {
+  const cache = new Map();
+  const checkSideEffectts = (filePath: string): boolean => {
```
Comment (askoufis) [inline packages/consumer/build.test.ts:11]:
```suggestion
  const checkSideEffects = (filePath: string): boolean => {
```

**Comment by askoufis** [comment]
Thoughts on moving this to a `fixtures` folder rather than putting it inside `packages`? I know it's private and will never be published anyway, but in terms of alignment with other repos (crackle, VE), I'd prefer it to be in `fixtures`, if it's not too much effort to do so.

**Comment by mrm007** [comment]
`fixtures` is a good idea 👍

**Review by askoufis — APPROVED** [review]

**Pull request merged**